core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -31,164 +35,182 @@ export interface QuestionBackendDict {
+    this._linkedSkillIds = linkedSkillIds;
+    this._inapplicableSkillMisconceptionIds = (
+      inapplicableSkillMisconceptionIds);
+  }
 
-    Question.prototype.getId = function() {
-      return this._id;
-    };
+  getId(): string {
+    return this._id;
+  }
 
-    Question.prototype.getStateData = function() {
-      return this._stateData;
-    };
+  getStateData(): State {
+    return this._stateData;
+  }
 
-    Question.prototype.setStateData = function(newStateData) {
-      this._stateData = angular.copy(newStateData);
-    };
+  setStateData(newStateData: State): void {
+    this._stateData = angular.copy(newStateData);
+  }
 
-    Question.prototype.getLanguageCode = function() {
-      return this._languageCode;
-    };
+  getStateDataSchemaVersion(): number {
+    return this._stateDataSchemaVersion;
+  }
 
-    Question.prototype.setLanguageCode = function(languageCode) {
-      this._languageCode = languageCode;
-    };
+  getLanguageCode(): string {
+    return this._languageCode;
+  }
 
-    Question.prototype.getVersion = function() {
-      return this._version;
-    };
+  setLanguageCode(languageCode: string): void {
+    this._languageCode = languageCode;
+  }
 
-    Question.prototype.getLinkedSkillIds = function() {
-      return this._linkedSkillIds;
-    };
+  getVersion(): number {
+    return this._version;
+  }
 
-    Question.prototype.setLinkedSkillIds = function(linkedSkillIds) {
-      this._linkedSkillIds = linkedSkillIds;
-    };
+  getLinkedSkillIds(): string[] {
+    return this._linkedSkillIds;
+  }
 
-    Question.prototype.getInapplicableSkillMisconceptionIds = function() {
-      return this._inapplicableSkillMisconceptionIds;
-    };
+  setLinkedSkillIds(linkedSkillIds: string[]): void {
+    this._linkedSkillIds = linkedSkillIds;
+  }
 
-    Question.prototype.setInapplicableSkillMisconceptionIds = function(
-        inapplicableSkillMisconceptionIds) {
-      this._inapplicableSkillMisconceptionIds = (
-        inapplicableSkillMisconceptionIds);
-    };
+  getInapplicableSkillMisconceptionIds(): string[] {
+    return this._inapplicableSkillMisconceptionIds;
+  }
 
-    // TODO(ankita240796): Remove the bracket notation once Angular2 gets in.
-    /* eslint-disable-next-line dot-notation */
-    Question['createDefaultQuestion'] = function(skillIds) {
-      return new Question(
-        null, StateObjectFactory.createDefaultState(null),
-        DEFAULT_LANGUAGE_CODE, 1, skillIds, []);
-    };
+  setInapplicableSkillMisconceptionIds(
+      inapplicableSkillMisconceptionIds: string[]): void {
+    this._inapplicableSkillMisconceptionIds = (
+      inapplicableSkillMisconceptionIds);
+  }
 
-    Question.prototype.getValidationErrorMessage = function() {
-      var interaction = this._stateData.interaction;
-      if (interaction.id === null) {
-        return 'An interaction must be specified';
+  getValidationErrorMessage(): string {
+    var interaction = this._stateData.interaction;
```

Comment:
Use `let/const` instead of `var` here and everywhere else. From the context, it looks like it can be const.

Reply:
Done!